In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
bank_df = pd.read_csv("../data/processed/bank_clean.csv")
customers_df = pd.read_csv("../data/processed/customer_clean.csv")

In [4]:
bank_df.head()

age        job  marital    education  default  housing   loan    contact  \
0  38.0  housemaid  MARRIED     basic.4y      0.0    False  False  telephone   
1  57.0   services  MARRIED  high.school  unknown    False  False  telephone   
2  37.0   services  MARRIED  high.school      0.0     True  False  telephone   
3  40.0     admin.  MARRIED     basic.6y      0.0    False  False  telephone   
4  56.0   services  MARRIED  high.school      0.0    False   True  telephone   

   duration  campaign  ...  cons.price.idx  cons.conf.idx euribor3m  \
0       261         1  ...          93.994          -36.4  4.857000   
1       149         1  ...          93.994          -36.4  3.626521   
2       226         1  ...          93.994          -36.4  4.857000   
3       151         1  ...          93.994          -36.4  3.625715   
4       307         1  ...          93.994          -36.4  3.581983   

   nr.employed   y        date  latitude longitude  \
0         5191  no  2019-08-02    41.495   -71.233   
1         5191  no  2016-09-14    34.601   -83.923   
2         5191  no  2019-02-15    34.939   -94.847   
3         5191  no  2015-11-29    49.041   -70.308   
4         5191  no  2017-01-29    38.033  -104.463   

                                    id_ contact_year  
0  089b39d8-e4d0-461b-87d4-814d71e0e079         2019  
1  e9d37224-cb6f-4942-98d7-46672963d097         2016  
2  3f9f49b5-e410-4948-bf6e-f9244f04918b         2019  
3  9991fafb-4447-451a-8be2-b0df6098d13e         2015  
4  eca60b76-70b6-4077-80ba-bc52e8ebb0eb         2017  

[5 rows x 24 columns]

In [5]:
customers_df.head()

Income  Kidhome  Teenhome Dt_Customer  NumWebVisitsMonth  \
0  161770        1         0  2012-04-04                 29   
1   85477        1         1  2012-12-30                  7   
2  147233        1         1  2012-02-02                  5   
3  121393        1         2  2012-12-21                 29   
4   63164        1         2  2012-06-20                 20   

                                    id_  Total_children  
0  089b39d8-e4d0-461b-87d4-814d71e0e079               1  
1  e9d37224-cb6f-4942-98d7-46672963d097               2  
2  3f9f49b5-e410-4948-bf6e-f9244f04918b               2  
3  9991fafb-4447-451a-8be2-b0df6098d13e               3  
4  eca60b76-70b6-4077-80ba-bc52e8ebb0eb               3

In [6]:
print(f"Bank dataframe has {bank_df["id_"].nunique()} unique id's and customers dataframe has {customers_df["id_"].nunique()}")


Bank dataframe has 42752 unique id's and customers dataframe has 43170


In [7]:
set_a = set(bank_df["id_"])
set_b = set(customers_df["id_"])
intersection = set_a.intersection(set_b)
len(intersection)

42752

After validating the unique identifiers in both datasets, it was confirmed that there are 42,752 common IDs between the bank campaign dataset and the customer demographic dataset.
Since all campaign records have a corresponding demographic record, an inner join will be performed based on the id_ column.
This ensures that no campaign data is lost during the integration process.

In [8]:
df_integrated = pd.merge(left=bank_df, right=customers_df, on="id_", how="inner")
df_integrated.head()

age        job  marital    education  default  housing   loan    contact  \
0  38.0  housemaid  MARRIED     basic.4y      0.0    False  False  telephone   
1  57.0   services  MARRIED  high.school  unknown    False  False  telephone   
2  37.0   services  MARRIED  high.school      0.0     True  False  telephone   
3  40.0     admin.  MARRIED     basic.6y      0.0    False  False  telephone   
4  56.0   services  MARRIED  high.school      0.0    False   True  telephone   

   duration  campaign  ...  latitude  longitude  \
0       261         1  ...    41.495    -71.233   
1       149         1  ...    34.601    -83.923   
2       226         1  ...    34.939    -94.847   
3       151         1  ...    49.041    -70.308   
4       307         1  ...    38.033   -104.463   

                                    id_  contact_year  Income  Kidhome  \
0  089b39d8-e4d0-461b-87d4-814d71e0e079          2019  161770        1   
1  e9d37224-cb6f-4942-98d7-46672963d097          2016   85477        1   
2  3f9f49b5-e410-4948-bf6e-f9244f04918b          2019  147233        1   
3  9991fafb-4447-451a-8be2-b0df6098d13e          2015  121393        1   
4  eca60b76-70b6-4077-80ba-bc52e8ebb0eb          2017   63164        1   

   Teenhome Dt_Customer NumWebVisitsMonth Total_children  
0         0  2012-04-04                29              1  
1         1  2012-12-30                 7              2  
2         1  2012-02-02                 5              2  
3         2  2012-12-21                29              3  
4         2  2012-06-20                20              3  

[5 rows x 30 columns]

In [9]:
df_integrated.to_csv("../data/processed/integrated.csv")